packages/components/src/components/dropdown/dropdown.tsx

```diff
@@ -325,7 +343,19 @@ const Dropdown = ({
         handleVisibility();
     };
 
-    const handleVisibility = () => {
+    const handleVisibility = (e?: React.MouseEvent<HTMLDivElement>) => {
+        if (should_open_on_hover) {
+            if (e?.type === 'mouseover') {
+                setIsListVisible(true);
+                return;
+            } else if (e?.type === 'mouseleave') {
+                setIsListVisible(false);
+                return;
+            }
+        } else if (e?.type === 'mouseover' || e?.type === 'mouseleave') {
+            return;
+        }
```

Comment:
```suggestion
        if (should_open_on_hover && ['mouseover', 'mouseleave'].includes(e?.type)) {
            if (e?.type === 'mouseover') setIsListVisible(true);
            else setIsListVisible(false);
            return;
        }
```
here e?.type can be only one of the 2 options (either mouseover or mouseleave) :)

Reply:
This is not working, as this function also react on click.